### PR TITLE
[RC01 Ch03] adds ViNePERF in section 3.3.7 

### PR DESCRIPTION
**Pull request by acmacm**
See issue cntt-n#2390

### CODE REVIEW

**Comment by collivier** [comment]
@acmacm the test case list is missing in the mandatory list.

**Comment by acmacm** [comment]
@collivier wrote:
> 
> @acmacm the test case list is missing in the mandatory test case list.

Thanks for your comment. Can you help me find the test case list you are referring to?  Thanks again...

File: doc/ref_cert/RC1/chapters/chapter03.md [hunk 1]
```diff
@@ -724,7 +724,15 @@ At the time of writing, no KPI is defined in CNTT chapters which would have
...
 
-On top of this dataplane benchmarking described in VMTP & Shaker, we need to integrate testing as described in [ETSI GS NFV-TST 009: Specification of Networking Benchmarks and Measurement Methods for NFVI](https://www.etsi.org/deliver/etsi_gs/NFV-TST/001_099/009/03.01.01_60/gs_NFV-TST009v030101p.pdf). This type of testing is better suited to measure the networking capabilities of a compute node. The [rapid scripts](https://wiki.opnfv.org/display/SAM/Rapid+scripting) in conjunction with the [PROX tool](https://wiki.opnfv.org/pages/viewpage.action?pageId=12387840) offers an open source implementation for this type of testing.
+Industry Standard-based Benchmarking Projects
+
+In addition to the dataplane testing described in Functest, two projects have integrated industry standard benchmarking test as described in [ETSI GS NFV-TST 009: Specification of Networking Benchmarks and Measurement Methods for NFVI](https://www.etsi.org/deliver/etsi_gs/NFV-TST/001_099/009/03.04.01_60/gs_NFV-TST009v030401p.pdf), and other IETF standards as well. 
+- [SampleVNF](https://wiki.anuket.io/display/HOME/Sample+VNF)
```
Comment (lylavoie) [inline doc/ref_cert/RC1/chapters/chapter03.md:730]: Didn't SampleVNF get renamed, or am I misremembering that?

File: doc/ref_cert/RC1/chapters/chapter03.md [hunk 2]
```diff
@@ -724,7 +724,15 @@ At the time of writing, no KPI is defined in CNTT chapters which would have
...
+- [ViNePERF](https://wiki.anuket.io/display/HOME/ViNePERF)
+
+This type of testing is better suited to measure the networking capabilities of a compute node. The [rapid scripts](https://wiki.opnfv.org/display/SAM/Rapid+scripting) in conjunction with the [PROX tool](https://wiki.opnfv.org/pages/viewpage.action?pageId=12387840) offers an open source implementation for this type of testing.
```
Comment (lylavoie) [inline doc/ref_cert/RC1/chapters/chapter03.md:733]: Should these Wiki pages be moved over to wiki.anuket.io?

**Review by lylavoie — APPROVED** [review]
only minor nits

File: doc/ref_cert/RC1/chapters/chapter03.md (hunk 1 above)
Comment (acmacm) [reply doc/ref_cert/RC1/chapters/chapter03.md:730]: Not yet, AFAIK.  VSPERF was re-named ViNePERF a few months back.  There was some discussion of combining the two projects.

File: doc/ref_cert/RC1/chapters/chapter03.md (hunk 2 above)
Comment (acmacm) [reply doc/ref_cert/RC1/chapters/chapter03.md:733]: Yes, but they have not been moved, I guess this is a resource issue. The links still work, and in a future commit we should update the status of SampleVNF.

**Review by trevgc — APPROVED** [review]

**Comment by collivier** [comment]
Sorry if I was unclear, your PR does also include the container and the test case name in 3.4.
Then it requires to add then in the RC playbook (I could help here) + the input files needed to execute the test in RC cookbook.

Could you print the daily job to evaluate the remaining work?
Daily master seems compiling soft rather than leveraging containers as mandatory in RC.
https://build.opnfv.org/ci/view/vswitchperf/job/vswitchperf-daily-master/664/console

We could init an adhoc meeting if it helps.

**Review by collivier — CHANGES_REQUESTED** [review]
Sorry if I was unclear, your PR does also include the container and the test case name in the mandatory test case list in 3.4.
Then it requires to add them in the RC playbook (I could help here) + the input files needed to execute the RC cookbook from scratch.

Could you print the daily job to evaluate the remaining work?
Daily master seems compiling soft rather than leveraging containers as mandatory in RC.
https://build.opnfv.org/ci/view/vswitchperf/job/vswitchperf-daily-master/664/console

We could init an adhoc meeting if it helps.

**Comment by acmacm** [comment]
Hi @collivier / Cedric,

I am trying to add the same level of description for the ViNePERF tests as was done for the SampleVNF project last year.  The short answer (below) is that the ViNePERF containerization requires some more work.

> Sorry if I was unclear, your PR does also include the container and the test case name in the mandatory test case list in 3.4.

When I check https://cntt-n.github.io/CNTT/doc/ref_cert/RC1/chapters/chapter03.html#3.4 , I don't see any SampleVNF tests in the mandatory test case list, so this is a first step for ViNePerf, similar to the material that SampleVNF provided/merged.

IIRC, Sridhar had VSPERF installing and running from a container, but there may have been a problem with running T-rex in another container (on another host). - It's something to check, and I know that Sridhar/  @opensource-tnbt  is busy bringing the Interns on-board now...

> Then it requires to add them in the RC playbook (I could help here) + the input files needed to execute the RC cookbook from scratch.
> 
> Could you print the daily job to evaluate the remaining work?
> Daily master seems compiling soft rather than leveraging containers as mandatory in RC.
> https://build.opnfv.org/ci/view/vswitchperf/job/vswitchperf-daily-master/664/console
> 
> We could init an adhoc meeting if it helps.

Thanks, that may help us soon!  It's a matter of scheduling - we lost too much time and now the content freeze is upon us. I think we could merge these commits on the same basis as SampleVNF's were, with more work to do to complete the material in section 3.4.

**Review by opensource-tnbt — APPROVED** [review]

**Comment by collivier** [comment]
You're true all refs to YardStick, Dovetail or software which are offering test cases would be removed before publishing to GSMA.
It's part of the technical debt.

I don't see the point to merge this patch if Vineperf goes to the technical debts later (hoping not ofc).
Yes we must go quickly here but I thought the Xtesting part was already done.

**Comment by acmacm** [comment]
Hi collivier/Cedric,
> 
> 
> You're true all refs to YardStick, Dovetail or any formet softwares which aren't offering any test case would be removed before publishing to GSMA (RC is not about publicity ;)).

I agree, "publicity" should be reduced in the futre, but there IS a plan to add the code from ViNePERF, unlike the others.

> It's badly part of the technical debt and the bazaar model at the beginning.
> 
> I don't see the point to merge this patch if Vineperf goes to the technical debts later (hoping not ofc).

For me, and probably others who commented, this PR started a learning process about how you would like the RC documentation to operate in future. So that's the positive value of this PR. 

> Yes we must go quickly here but I thought the Xtesting part was already done.

opensource-tnbt / Sridhar reminded me that the Xtesting integration accomplished for JERMA more like a proof-of-concept, and there was about 2-3 days more work to match the software with the test-list described here (in this PR for 3.3.7).

That work is ahead of us before final Kali release. Meanwhile, I think we have delivered what we promised in this PR for 3.3.7, and we should enter a new PR for CH03 section 3.4 where you asked for additional changes !

**Comment by acmacm** [comment]
Yes, I was trying to close a Change request as out of scope (3.4 wanted, scope is 3.3.7). The PR is ready to Merge, IMO, with 3 reviews, etc.

**Comment by wmk-admin** [comment]
@collivier and @acmacm, please advise (both) whether the agreement have been reached and / or what the next actions are needed.

**Comment by collivier** [comment]
@wmk-admin hoping this quick summary makes my point clearer.
ViNePERF could be advertised in 4.2 as Functest, YardStick, Bottleneck, etc.
There is no test case conformed with the RC rules and we can hardly describe a test case which cannot be executed via the official playbook.
4.2 will be cleant as soon as the new GSMA starts (see YardStick, Bottleneck, etc.).

I can't evaluate the gap to integrate ViNePerf in RC1 as there is nothing clear in Jenkins.
I know there is a strong misunderstanding between static milestones and gates/daily jobs but here I do emphasize that chapter 3 is about what's running via RC1.

Next actions:
 - xtesting-based test cases
 - all config files described in cookbook
 - daily jobs in Jenkins highlighting the true the test case behavior and its quality (OPNFV has loved false positives)
 - describing the test cases and the requirement coverage.

@walterkozlowski @acmacm 
You could let the PR open (I would have put the L-label), but I would rather suggest to start a new PR for 4.2.
Chapter 3 asks for lots of automation improvements which, from my understanding, can't be achieved  before 1st July (not taken into account M5).

**Review by collivier — CHANGES_REQUESTED** [review]
See comments

**Comment by pgoyal01** [comment]
@acmacm @collivier @walterkozlowski Next Steps?

**Pull request closed without merging**